extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/ValidatorFactoryCustomizer.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.hibernate.validator;
+
+import org.hibernate.validator.BaseHibernateValidatorConfiguration;
+
+/**
+ * Meant to be implemented by a CDI bean that provides arbitrary customization for the default
+ * {@link javax.validation.ValidatorFactory}.
+ * <p>
+ * All implementations that are registered as CDI beans are taken into account when producing the default
+ * {@link javax.validation.ValidatorFactory}.
+ * <p>
+ * Customizers are applied in the order of {@link javax.annotation.Priority}.
+ */
+public interface ValidatorFactoryCustomizer {
+
+    void customize(BaseHibernateValidatorConfiguration<?> configuration);
```

Comment:
I'm not sure removing `Hibernate` from the class name of the customizer is such a good idea, since the configuration argument to the `customize` method **is** specific to Hibernate Validator.

IMO, if you're going to remove `Hibernate` from the class name of the customizer, then the `customize` method should accept a `Configuration<?>` (which is not specific to Hibernate Validator). Which would remove plenty of configuration options.

So personally, I wouldn't remove `Hibernate` from the class name... :shrug:

Reply:
Well, people are manipulating `ValidatorFactory` so I prefer keeping that name for the customizer. We are using this pattern everywhere.